File: leetcode/medium/Q34_FindFirstandLastPositionofElementinSortedArray.py
from typing import List

class Solution:
    def searchRange(self, nums: List[int], target: int) -> List[int]:
        start = 0
        end = len(nums) - 1
        sflag, eflag = False, False

        while start <= end and (not sflag or not eflag):
            if nums[start] != target and not sflag:
                start += 1
            else:
                sflag = True
            if nums[end] != target and not eflag:
                end -= 1
            else:
                eflag = True
        if start >= len(nums) or end < 0:
            return [-1, -1]

        if nums[start] != target:
            start = -1
        if nums[end] != target:
            end = -1
        return [start, end]

File: src/algorithm1/binarysearch/Q34_FindFirstandLastPositionofElementinSortedArray.java
package algorithm1.binarysearch;

public class Q34_FindFirstandLastPositionofElementinSortedArray {
    class Solution {
        int[] answer = {-1, -1};
        public int[] searchRange(int[] nums, int target) {
            answer[0] = -1;
            answer[1] = -1;
            find(0, nums.length, nums ,target);
            return answer;
        }

        public void find(int start, int end, int[] nums, int target){
            if(start < end){
                int mid = (start + end) / 2;
                if(nums[mid] == target){
                    System.out.println(mid);
                    answer[0] = answer[0] == - 1 ? mid : Math.min(answer[0], mid);
                    answer[1] = answer[1] == - 1 ? mid : Math.max(answer[1], mid);
                }

                if(nums[mid] >= target){
                    find(start, mid, nums, target);
                }

                if(nums[mid] <= target){
                    find(mid + 1, end, nums, target);
                }
            }
        }
    }
}
